token/program-2022/src/extension/mod.rs

```diff
@@ -836,24 +755,135 @@ impl<'data, S: BaseState> StateWithExtensionsMut<'data, S> {
     /// state initialization
     /// Noops if there is no room for an extension in the account, needed for
     /// pure base mints / accounts.
-    pub fn init_account_type(&mut self) -> Result<(), ProgramError> {
-        if !self.account_type.is_empty() {
-            if let Some(extension_type) = self.get_first_extension_type()? {
+    fn init_account_type(&mut self) -> Result<(), ProgramError> {
+        let first_extension_type = self.get_first_extension_type()?;
```

Comment:
Why did you move this expression? It seems heavier than getting account_type and checking whether it's empty.

Reply:
We're fetching the account type, which means borrowing `self` as mutable, and holding the resulting reference, so we can't call `self` as immutable in `get_first_extension_type`, so we get a compilation error if we don't move the call.

Maybe I was being a bit lazy, so if you can think of a slicker way to do this, I'm all for it!